### ISSUE TITLE
Hot reloading: Automatically allocate a free port for rsync sidecar.

### ISSUE THREAD

**Issue by thsig** [garden-io/garden#333]
This is a leftover from the big `hot-reload` PR (https://github.com/garden-io/garden/pull/291) that was merged today.

Currently, the rsync sidecar container listens on a fixed port (873), and the framework prevents the hot-reload enabled service (which the sidecar is attached to) from using port 873 as a `containerPort`.

This is obviously not ideal, since a free port (not used by the service) could be automatically allocated.

**Comment by edvald** [comment]
This should be slightly easier to fix after #422, because we no longer expose the sidecar on a `nodePort`.

**Comment by drubin** [comment]
I think we should mark this as invalid as won't fix. 

Randomly picking a freeport will cause a world of pain and many times will only be able to be detected at runtime. 

Rsync ports is a standard port and the chances of collisions is almost zero unless some one is working on running Rsync server inside their container.

**Comment by drubin** [comment]
I am going to close this now feel free to open if you feel strongly we should spend time on this.

873 is officially reserved for Rsync.
